Modules/StudyProgramme/classes/class.ilObjStudyProgramme.php

```diff
@@ -55,36 +60,63 @@ public function __construct($a_id = 0, $a_call_by_reference = true) {
 		$this->object_factory = ilObjectFactoryWrapper::singleton();
 		self::initStudyProgrammeCache();
 	}
-	
+
 	static public function initStudyProgrammeCache() {
 		if (self::$study_programme_cache === null) {
 			self::$study_programme_cache = ilObjStudyProgrammeCache::singleton();
 		}
 	}
-	
+
+	/**
+	* Get a (cached) instance of ilStudyProgrammeUserProgressDB
+	*
+	* @return ilStudyProgrammeUserProgressDB
+	*/
+	public function getStudyProgrammeUserProgressDB() {
+		if(! $this->sp_user_progress_db) {
+			$this->sp_user_progress_db = static::getNewStudyProgrammeUserProgressDB();
+		}
+		return $this->sp_user_progress_db;
+	}
+
+	/**
+	* Get a new instance of ilStudyProgrammeUserProgressDB
+	*
+	* @return ilStudyProgrammeUserProgressDB
+	*/
+	static public function getNewStudyProgrammeUserProgressDB() {
```

Comment:
The name also contains the missleading `New`. To avoid a nameclash with the non-static method we could use an (ugly) "_" in front of the name...